### PR TITLE
Create algorithm training module, fs_algo

### PR DESCRIPTION
**Pull request by glitt13**
Performs algorithm training and basic test evaluation, where basin attribute data predict model performance and/or hydrologic signatures

## Additions

python fs_algo module

## Removals

-

## Changes

-

## Testing

Unit testing in fs_algo module's fs_algo_train_eval.py at 92% coverage

## Notes

-

## Todos

 - Consider scikit-learn grid search approach to multiple algorithms for future design improvement/simplification

## Checklist

- [ ] PR has an informative and human-readable title
- [ ] Changes are limited to a single goal (no scope creep)
- [ ] Code can be automatically merged (no conflicts)
- [ ] Code follows project standards (link if applicable)
- [ ] Passes all existing automated tests
- [ ] Any _change_ in functionality is tested
- [ ] New functions are documented (with a description, list of inputs, and expected output)
- [ ] Placeholder code is flagged / future todos are captured in comments
- [ ] Visually tested in supported browsers and devices (see checklist below :point_down:)
- [ ] Project documentation has been updated (including the "Unreleased" section of the CHANGELOG)
- [ ] Reviewers requested with the [Reviewers tool](https://help.github.com/articles/requesting-a-pull-request-review/) :arrow_right:

### Target Environment support

- [ ] Windows
- [ ] Linux
- [ ] Browser

### Accessibility

- [ ] Keyboard friendly
- [ ] Screen reader friendly

### Other

- [ ] Is useable without CSS
- [ ] Is useable without JS
- [ ] Flexible from small to large screens
- [ ] No linting errors or warnings
- [ ] JavaScript tests are passing

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -25,6 +25,13 @@ Describe any dependencies that must be installed for this software to work.
...
+- [NOAA-OWP/hydrofabric](https://github.com/NOAA-OWP/hydrofabric)
+  - Note that the arrow package needs `arrow::arrow_with_s3() == TRUE`. If `FALSE`, consider downloading arrow via [apache's r-universe](https://apache.r-universe.dev/arrow)
+  - Steps to install hydrofabric: Refer to wiki
+- [USGS nhdplusTools](https://github.com/doi-usgs/nhdplusTools/)
+- [pynhd](https://github.com/hyriver/pynhd)
+
+
 ## Installation - proc_fsds python package
 
 ### TLDR
```
Comment (bolotinl) [inline README.md:37]: Note that this section has some inconsistencies, sometimes referring to the package as `proc_fsds`, but the package is actually called `fsds_proc`.

**Comment by bolotinl** [comment]
The code works (with the minor suggested changes) and creates a trained ML algorithm. More fine-detailed review may be desired later.

**Review by bolotinl — APPROVED** [review]
The code works (with the minor suggested changes) and creates a trained ML algorithm output (one parquet file and two joblib files). More fine-detailed review may be desired later.

File: pkg/fs_algo/fs_algo/fs_proc_algo.py
[the remark this answers is not in the document]
Comment (glitt13) [reply pkg/fs_algo/fs_algo/fs_proc_algo.py:1]: These 'scripts' should generally be automated under the assumption that the data and configuration files have already been standardized into a common format, so they'll live close to the package that they use. We may make them part of a package eventually.

In the case of the metrics input dataset munging, those are one-off scripts unique to each metrics or hyd sig dataset, so we'll try and keep those farther away from the packages.

File: README.md (hunk 1 above)
Comment (glitt13) [reply README.md:37]: @bolotinl - addressed in the latest commit.

**Review by bolotinl — APPROVED** [review]
